index.js

```diff
@@ -50,59 +44,95 @@ function getContent(unix, preferUsability = false) {
-				await interaction.editReply({
-					content: getContent(unix, preferUsability)
-				});
-				console.log(getConsoleContent(false, ephemeral, day, month, year, hour, minute, second, timezone));
-				break;
-			case 'currenttimestamp':
-				await interaction.deferReply({ ephemeral });
-				unix = DateTime.now().toUnixInteger();
-				await interaction.editReply({
-					content: getContent(unix, preferUsability)
-				});
-				console.log(getConsoleContent(true, ephemeral));
-				break;
-		}
-	} else if (interaction.isAutocomplete()) {
-		let timezoneResponse = Intl.supportedValuesOf('timeZone').filter(zone => {
-			return zone.toLowerCase().indexOf(interaction.options.getFocused().replace(' ', '_').toLowerCase()) >= 0;
-		});
-		timezoneResponse.length = Math.min(timezoneResponse.length, 25); // send max. 25 choices
-		await interaction.respond(timezoneResponse.map(zone => {
-			return {
-				name: zone,
-				value: zone
-			};
-		})).catch(err => console.error(JSON.stringify(err)));
-	}
+client.on("interactionCreate", async (interaction) => {
+  if (interaction.isCommand()) {
+    let preferUsability =
+      interaction.options.getBoolean("prefer_usability") ?? false;
+    let ephemeral = !interaction.options.getBoolean("public");
+    let unix = 0;
+    switch (interaction.commandName) {
+      case "timestamp":
+        await interaction.deferReply({ ephemeral });
+        let day = interaction.options.getInteger("day") ?? 1;
+        let month = interaction.options.getInteger("month") ?? 1;
+        let year = interaction.options.getInteger("year") ?? 1970;
+        let hour = interaction.options.getInteger("hour") ?? 0;
+        let minute = interaction.options.getInteger("minute") ?? 0;
+        let second = interaction.options.getInteger("second") ?? 0;
+        let timezone = interaction.options.getString("timezone");
+        //Intl.DateTimeFormat().resolvedOptions().timeZone // own timezone
+        unix = DateTime.now()
+          .setZone(timezone)
+          .set({ day, month, year, hour, minute, second })
+          .toUnixInteger();
+        await interaction.editReply({
+          content: getContent(unix, preferUsability),
+        });
+        console.log(
+          getConsoleContent(
+            false,
+            ephemeral,
+            day,
+            month,
+            year,
+            hour,
+            minute,
+            second,
+            timezone,
+          ),
+        );
+        break;
+      case "currenttimestamp":
+        await interaction.deferReply({ ephemeral });
+        unix = DateTime.now().toUnixInteger();
+        await interaction.editReply({
+          content: getContent(unix, preferUsability),
+        });
+        console.log(getConsoleContent(true, ephemeral));
+        break;
+    }
+  } else if (interaction.isAutocomplete()) {
+    let timezoneResponse = Intl.supportedValuesOf("timeZone").filter((zone) => {
```

Comment:
TODO: See if this value can be cached and only retrieved at startup. Also check if it makes a difference if it is cached during startup or retrieving it each time.